### PR TITLE
Allow modification of ListTile's horizontalTitleGap, minVerticalPadding, minLeadingWidth

### LINKED ISSUE #47647: Allow modification of ListTile's horizontalTitleGap
## Use case

This field is a const in ListTile:
```
 // The horizontal gap between the titles and the leading/trailing widgets
  static const double _horizontalTitleGap = 16.0;
```

## Proposal
But maybe i do not need this gap, or need it smaller. Can you make this field as a property that can be changed?

### PR DESCRIPTION
**Pull request by Playhi**
## Description

Allow the modification of ListTile's default horizontalTitleGap, minVerticalPadding, minLeadingWidth.

## Related Issues

* fixes https://github.com/flutter/flutter/issues/47647
* https://github.com/flutter/flutter/issues/28924

## Tests

I added the following tests:

* Added test `ListTile horizontalTitleGap = 0.0`
* Added test `ListTile horizontalTitleGap = (default)`
* Added test `ListTile minVerticalPadding = 80.0`
* Added test `ListTile minLeadingWidth = (default)`
* Added test `ListTile minLeadingWidth = 60.0`

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by xilibro** [comment]
More than half a year has passed... #47647

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -17,6 +17,18 @@ import 'material_state.dart';
 import 'theme.dart';
 import 'theme_data.dart';
 
+/// Define some default values for convenient external use.
+class ListTileDefaultValue {
```
Comment (darrenaustin) [inline packages/flutter/lib/src/material/list_tile.dart:21]: While I like the idea here, it doesn't match the pattern used elsewhere in the code base. To be consistent, can you just use these values directly in the constructor and remove this public class?

**Review by darrenaustin — CHANGES_REQUESTED** [review]
@Playhi thanks so much for the contribution. My apologies on not getting to this sooner. It looks good. I just have a quibble with the introduction of a new defaults class interface, but otherwise it looks great.

File: packages/flutter/lib/src/material/list_tile.dart (hunk 1 above)
Comment (Playhi) [reply packages/flutter/lib/src/material/list_tile.dart]: Moved, and now it is directly in the constructor.

**Review by darrenaustin — APPROVED** [review]
LGTM. Thanks for the contribution.

**Comment by darrenaustin** [comment]
@Playhi, unfortunately there was a recent change to some community tests dealing with package versioning. This is throwing off the tests for this PR. Can you merge with the latest from master and push to this PR?

The easiest way to do this is something like:

```bash
git pull --rebase upstream master
git push --force origin list_tile_more_customizable
```

Hopefully that will clear up the tests and we can land this change.

**Comment by darrenaustin** [comment]
@Playhi thanks for the update, but it looks like there is a still a merge conflict with what is on master. Perhaps someone else checked in something conflicting since your update? Can you merge one more time?

**Pull request merged**